pycoal/mineral.py

```diff
@@ -2,182 +2,197 @@
+        # we may need to load it as an image instead
+        library = spectral.io.envi.open(libraryFileName)
+
+        # get dimensions of library to create mask for training class
+        dimensions = library.spectra.shape
+
+        # create mask
+        mask = [[x for x in range(dimensions[1])] for y in range(dimensions[0])]
+
+        # generate training data
+        trainingData = spectral.algorithms.create_training_classes(self.normalize(library), mask)
+
+        # initialize and train a neural network using the training data
+        classifier = spectral.classifiers.PerceptronClassifier(???)
+        classifier.train(trainingData, ???)
+
+        # return trained classifier
+        return classifier
+
+    def saveClassifier(self, classifier, classifierFilename):
+        """
+        This saves a classifier to a file for reuse.
+        Args:
+           classifier (PerceptronClassifier): the classifier to write
+           classifierFilename (str): the file which to write the classifier
+        Returns:
+            None
+        """
+
+        with open(classifierFilename,"wb") as f:
+            pickle.dump(classifier, f)
+
+    def readClassifier(self, classifierFilename):
+        """
+        This reads a classifier from a file.
+        Args:
+           classifierFilename (str): the file from which to load the classifier
+        Returns:
+            ??? not sure about how to express this
+        """
+
+        with open(classifierFilename, "rb") as f:
+            return pickle.load(f)
+
+    def classifyPixel(self, pixel, classifier, wavelengthList):
+        """
+        Return the class id of an AVIRIS pixel classified using a trained classifier.
+        Args:
+            pixel (???): the pixel to be classified
+           classifier (PerceptronClassifier): the classifier
+           wavelengthList (list): the list of wavelengths for the pixel
+        Returns:
+            Class ID of pixel
+        """
+        # classify normalized pixel and return class id
+        return classifier.classify_spectrum(self.normalize(pixel, wavelengthList))
+
+    def classifyImage(self, imageFilename, classifier, classifiedImageFilename):
+        """
+        Classify an AVIRIS image using a trained classifier and save the results to a file.
+        Args:
+           imageFilename (str): the image to be classified
+           classifier (PerceptronClassifier): the classifier
+           classifiedImageFilename (str): the file which to save the classified image
+        Returns:
+            None
+        """
+
+        # open the image
+        image = spectral.io.aviris.open_image(imageFilename)
+
+        # change units from nanometers to micrometers if applicable
+        if image.metadata.get('wavelength units') == u'Nanometers':
+            image.metadata['wavelength'] = [x / 1000 for x in image.metadata.get('wavelength')]
+            image.metadata['wavelength units'] = u'Micrometers'
```

Comment:
There's a small difference, but I'm pretty sure we don't have to worry about the wavelength and unit keys not existing so it's not crucial to our code. 

See this post: http://stackoverflow.com/questions/11041405/why-dict-getkey-instead-of-dictkey